homeassistant/components/cover/zwave.py

```diff
@@ -110,24 +111,43 @@ class ZwaveGarageDoor(zwave.ZWaveDeviceEntity, CoverDevice):
     def __init__(self, values):
         """Initialize the zwave garage door."""
         ZWaveDeviceEntity.__init__(self, values, DOMAIN)
+        self._door_state_list = None
+        self._door_state = None
         self.update_properties()
 
     def update_properties(self):
         """Handle data changes for node values."""
-        self._state = self.values.primary.data
+        self._door_state = self.values.primary.data
+        door_state_list = self.values.primary.data_items
+        if door_state_list:
+            self._door_state_list = list(door_state_list)
```

Comment:
Exactly. We already have open and close, so we just need to add these new states.

Then, you'll have something like this in zwave.py
```python
ZWAVE_STATES = {
    "Open": STATE_OPEN,
    "Closed": STATE_CLOSED,
    ...
}

...

self._door_state = ZWAVE_STATES.get(self.values.primary.data)
```

It seems excessive now, but we need to have hass specific constants for when other platforms implement this interface.